java/test/src/main/java/io/ray/test/CrossLanguageInvocationTest.java

```diff
@@ -188,6 +189,85 @@ public void testPassActorHandleFromJavaToPython() {
     Assert.assertEquals(res.get(), "3".getBytes());
   }
 
+  @Test
+  public void testExceptionSerialization() {
+    try {
+      throw new NativeRayException("Test Exception");
+    } catch (NativeRayException e) {
+      try {
+        String formattedException = org.apache.commons.lang3.exception.ExceptionUtils
+            .getStackTrace(e);
+        RayException exception = RayException.parseFrom(e.toBytes());
+        Assert.assertEquals(exception.getFormattedExceptionString(), formattedException);
+      } catch (InvalidProtocolBufferException ex) {
+        ex.printStackTrace();
+        Assert.fail();
+      }
+    }
+  }
+
+  @Test
+  public void testRaiseExceptionFromPython() {
+    ObjectRef<Object> res = Ray.task(PyFunction.of(
+        PYTHON_MODULE, "py_func_python_raise_exception", Object.class)).remote();
+    try {
+      res.get();
+    } catch (RuntimeException ex) {
+      // ex is a Python exception(py_func_python_raise_exception) with no cause.
+      Assert.assertTrue(ex instanceof CrossLanguageException);
+      CrossLanguageException e = (CrossLanguageException) ex;
+      Assert.assertEquals(e.getLanguage(), Language.PYTHON);
+      // ex.cause is null.
+      Assert.assertNull(ex.getCause());
+      Assert.assertTrue(ex.getMessage().contains("ZeroDivisionError: division by zero"),
```

Comment:
It seems that this only makes sure the root cause is `ZeroDivisionError`. Can we use a more strict assertion? I hope we can verify the type and message of every layer.

And the same for other tests.

Reply:
The problem is that in this solution, we can only get one level `cause` of cross language exception in Java. Suppose there is an exception chain: `Java -> Python -> Java`, Python can only concat the string of inner Java exception, and the outer Java exception will get one level `cause`: the Python exception.

I will add more strict assertions, but the assertions are on the exception string not the `cause` exception object.

Reply:
I wasn't expecting that you can assert on nested objects. Asserting on strings looks good to me.